mito-ai/src/Extensions/AiChat/ChatTaskpane.tsx

```diff
@@ -473,59 +478,148 @@ const ChatTaskpane: React.FC<IChatTaskpaneProps> = ({
+    ): Promise<boolean> => {
         setLoadingAIResponse(true)
 
-        try {
-            const aiResponse = await websocketClient.sendMessage<ICompletionRequest, ICompletionReply>(completionRequest);
+        if (completionRequest.stream) {
+            // Reset the streaming response and set streaming state
+            streamingContentRef.current = '';
+            
+            // Disconnect any existing stream handler
+            if (streamHandlerRef.current) {
+                websocketClient.stream.disconnect(streamHandlerRef.current, null);
+                streamHandlerRef.current = null;
+            }
 
-            if (aiResponse.error) {
-                console.error('Error calling OpenAI API:', aiResponse.error);
-                addAIMessageFromResponseAndUpdateState(
-                    aiResponse.error.hint
-                        ? aiResponse.error.hint
-                        : `${aiResponse.error.error_type}: ${aiResponse.error.title}`,
-                    completionRequest.metadata.promptType, // #TODO: Why are we storing the prompt type in the AI response?
-                    newChatHistoryManager,
-                    true,
-                    aiResponse.error.title
+            // Create the stream handler function and store it in the ref
+            const streamHandler = (
+                _: CompletionWebsocketClient, chunk: ICompletionStreamChunk
+            ) => {                
+                // Use a ref to accumulate the content properly
+                streamingContentRef.current += chunk.chunk.content;
+                
+                // Force a re-render to update the UI
+                forceUpdate();
+
+                // Update chat history with the complete accumulated content
+                newChatHistoryManager.addStreamingAIMessage(
+                    streamingContentRef.current, 
+                    completionRequest.metadata.promptType,
                 );
-            } else {
-                const content = aiResponse.items[0]?.content ?? '';
+                setChatHistoryManager(newChatHistoryManager);
+            };
+            
+            // Store the handler for later cleanup
+            streamHandlerRef.current = streamHandler;
+            
+            // Connect the handler
+            websocketClient.stream.connect(streamHandler, null);
 
-                if (completionRequest.metadata.promptType === 'agent:execution' || completionRequest.metadata.promptType === 'agent:autoErrorFixup') {
-                    // Agent:Execution prompts return a CellUpdate object that we need to parse
-                    const agentResponse: AgentResponse = JSON.parse(content)
-                    newChatHistoryManager.addAIMessageFromAgentResponse(agentResponse)
+            try {
+                const aiResponse = await websocketClient.sendMessage<ICompletionRequest, ICompletionReply>(completionRequest);
```

Comment:
When we're streaming, this aiResponse is just the very first chunk of the response. So what we're doing here is:

1. Get the first response from the AI 
2. Save it in the ChatHistory Manager 
3. In the streamHandler when we call the `addStreamingAIMessage` each time we get a chunk back, we overwrite the last AI message with the new content. 

I think a simple change that we could make here is just:

Once we call `await websocketClient.sendMessage`, we do error handling incase the connection fails, but then we don't do any of the message chatHistory stuff and let the ChatHistory Manager  just create the message. 

Does that make sense?